Content.Server/Body/Systems/SkeletonBodyManagerSystem.cs

```diff
@@ -0,0 +1,187 @@
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly ConstructionSystem _construction = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<SkeletonBodyManagerComponent, GetVerbsEvent<AlternativeVerb>>(AddReassembleVerbs);
+            SubscribeLocalEvent<ReassembleDoAfterComplete>(Reassemble);
+        }
+
+        /// <summary>
+        /// Adds the custom verb for reassembling skeleton parts
+        /// into a full skeleton
+        /// </summary>
+        private void AddReassembleVerbs(EntityUid uid, SkeletonBodyManagerComponent component, GetVerbsEvent<AlternativeVerb> args)
+        {
+            // every time i get warned of a null refernce i add another if statement
+            if (!args.CanAccess || !args.CanInteract)
+                return;
+
+            if (!TryComp<ActorComponent?>(args.User, out var actor))
+                return;
+
+            if (!TryComp<MindComponent>(uid, out var mind) || !mind.HasMind)
+                return;
+
+            var doAfterTime = component.DoAfterTime;
+            // doubles the time if you reconstruct yourself
+            if (args.User == uid)
+                doAfterTime *= 2;
+
+            // Custom verb
+            AlternativeVerb custom = new();
+            custom.Text = Loc.GetString("skeleton-reassemble-action");
+            custom.Act = () =>
+            {
+                if (!GetNearbyParts(args.User, uid, component, out var partList))
+                    return;
+
+                if (partList == null)
+                    return;
+
+                var doAfterEventArgs = new DoAfterEventArgs(component.Owner, doAfterTime, default, component.Owner)
+                {
+                    BreakOnTargetMove = true,
+                    BreakOnUserMove = true,
+                    BreakOnDamage = true,
+                    BreakOnStun = true,
+                    NeedHand = false,
+                    BroadcastFinishedEvent = new ReassembleDoAfterComplete(uid, args.User, component, partList),
+                };
+
+                _doAfterSystem.DoAfter(doAfterEventArgs);
+            };
+            custom.IconTexture = "/Textures/Mobs/Species/Skeleton/parts.rsi/full.png";
+            custom.Priority = 1;
+            args.Verbs.Add(custom);
+        }
+
+        private bool GetNearbyParts(EntityUid user, EntityUid uid, SkeletonBodyManagerComponent component, out List<EntityUid>? partList)
+        {
+            partList = new List<EntityUid>();
+
+            if (component.BodyParts == null)
+                return false;
+
+            // Ensures all of the old body part pieces are there
+            var nearby = _construction.EnumerateNearby(user);
+            bool notFound;
+            foreach (var part in component.BodyParts)
+            {
+                notFound = true;
+                foreach (var entity in nearby)
```

Comment:
i feel like these loops should be the other way around, iterate over nearby entities and check if any of them are parts you need. doesn't super matter since this doesnt exactly happen often anyway but

Reply:
I just made it check for distance for each prat given we don't care what else is nearby.